tests/test_kubernetes_tools.py

```diff
@@ -1062,11 +1073,73 @@ def test_get_pod_template_spec_smartstack_service(self):
                     containers=mock_get_kubernetes_containers.return_value,
                     share_process_namespace=True,
                     node_selector={"yelp.com/pool": "default"},
+                    affinity=V1Affinity(
+                        node_affinity=mock_get_node_affinity.return_value,
+                    ),
                     restart_policy="Always",
                     volumes=[],
                 ),
             )
 
+    @pytest.mark.parametrize(
```

Comment:
nit: I personally prefer three tests rather than using parametrize. This is a little hard to read for me. Also, by doing that, the extra bonus is you can use test name to point out what you really want to test, so no comments needed.